### DISCUSSION TITLE
Powers of tau trusted setup for PLONK over BW6-761

### DISCUSSION THREAD

**Discussion by secure12** [Consensys/gnark#922] [Q&A]
First of all, thank you for the great work!

We are trying to create proofs where BLS12-377 operations are involved (since 377 is the only pairing-friendly BLS curve implemented with also have an outer curve, namely BW6-761 implemented). However, we struggle to find a universal setup for PLONK over the outer curve BW6-761.

By any chance, do you know of any trusted setup done for BW6-761 that is publicly available?

**Comment by ivokub** [comment]
cc @yelhousni

**Reply by yelhousni** [reply]
There are actually 2 publicly available Groth16 trusted MPC setup on BW6-761. They are divided in two phases. Phase 1 corresponds to the powers of tau in Groth16 setup which you can use as a Plonk (KZG) setup. The first ceremony was conducted by Celo (https://blog.celo.org/the-plumo-ceremony-ac7649e9c8d8) and the second one by Aleo (https://setup.aleo.org/). You can find the transcript for Celo (2^27 points) [here](https://console.cloud.google.com/storage/browser/plumoceremonyphase1;tab=objects?project=crypto-plumo-files&prefix=&forceOnObjectsSortingFiltering=false) and Aleo (2^20 points *I believe*) [here](https://setup-staging.aleo.org/transcripts).

**Reply by secure12** [reply]
Thank you for these links.

We had previously identified Aleo and Plumo/Celo as relevant projects but had not found the links to the set-up outputs, so this is useful. Unfortunately, there seems to be permissions required to see the Plumo set-up, and we are unable to gain access. For Aleo, is there a place we can read about the precise definitions of the inner, outer, and universal set-ups? We expect that for a PLONK set-up, we care about the unviersal set-up only, but we would like to understand the difference between these three better.

Many thanks.

**Reply by yelhousni** [reply]
![Screenshot 2023-11-20 at 12 33 17](https://github.com/Consensys/gnark/assets/16170090/1cf29941-c8e3-483e-a728-516165eda263)
The files are public but somehow the interface isn't indeed. Anyway here is a command to download the files through `gsutil`:
```bash
gsutil -m cp \
  "gs://plumoceremonyphase1/chunks/0.0.0" \
  "gs://plumoceremonyphase1/chunks/0.0.1.0x061a61d29fa741f5444f53b038d83421b6bf843f" \
  "gs://plumoceremonyphase1/chunks/0.0.1.0x061a61d29fa741f5444f53b038d83421b6bf843f-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.0.1.0x16a3e97b7b3c04d885b15ff01c346349f936caba" \
  "gs://plumoceremonyphase1/chunks/0.0.1.0x16a3e97b7b3c04d885b15ff01c346349f936caba-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.0.10.0xb1a0d4cc4ee046a016d400f86d3f9036a6f6b192" \
  "gs://plumoceremonyphase1/chunks/0.0.10.0xb1a0d4cc4ee046a016d400f86d3f9036a6f6b192-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.0.11.0x76e27aebe37adc4f298047ad595150baef28ef38" \
  "gs://plumoceremonyphase1/chunks/0.0.11.0x76e27aebe37adc4f298047ad595150baef28ef38-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.0.12.0x9d5ac8a8a03ecc3f6f03e5383cafed04d37d3c9c" \
  "gs://plumoceremonyphase1/chunks/0.0.12.0x9d5ac8a8a03ecc3f6f03e5383cafed04d37d3c9c-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.0.2.0xa9ec7ffee66bab537d193fc41904fa2db9aac81e" \
  "gs://plumoceremonyphase1/chunks/0.0.2.0xa9ec7ffee66bab537d193fc41904fa2db9aac81e-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.0.2.0xe9053647d4bba6c53a88d1d012e0cf1af86c049f" \
  "gs://plumoceremonyphase1/chunks/0.0.2.0xe9053647d4bba6c53a88d1d012e0cf1af86c049f-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.0.3.0x3d776077c2ce34e258ab871a47698ec9a2c81251" \
  "gs://plumoceremonyphase1/chunks/0.0.3.0x3d776077c2ce34e258ab871a47698ec9a2c81251-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.0.4.0x44218a24cfdf0dba912ee5789d8de7a9382739b7" \
  "gs://plumoceremonyphase1/chunks/0.0.4.0x44218a24cfdf0dba912ee5789d8de7a9382739b7-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.0.5.0x823b3d45f7ec13a7de9f05f8174ef87a0ff37081" \
  "gs://plumoceremonyphase1/chunks/0.0.5.0x823b3d45f7ec13a7de9f05f8174ef87a0ff37081-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.0.6.0x44218a24cfdf0dba912ee5789d8de7a9382739b7" \
  "gs://plumoceremonyphase1/chunks/0.0.6.0x44218a24cfdf0dba912ee5789d8de7a9382739b7-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.0.7.0xf17076ace7117299eb7270f7c6d3181b8d0c7bc3" \
  "gs://plumoceremonyphase1/chunks/0.0.7.0xf17076ace7117299eb7270f7c6d3181b8d0c7bc3-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.0.8.0xce7ac5644fe2451bca7cdafc34f97f1637697393" \
  "gs://plumoceremonyphase1/chunks/0.0.8.0xce7ac5644fe2451bca7cdafc34f97f1637697393-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.0.9.0xf7c4330068e4ebaac4bb11d261b5219074cbdac8" \
  "gs://plumoceremonyphase1/chunks/0.0.9.0xf7c4330068e4ebaac4bb11d261b5219074cbdac8-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.1.0" \
  "gs://plumoceremonyphase1/chunks/0.1.1.0x061a61d29fa741f5444f53b038d83421b6bf843f" \
  "gs://plumoceremonyphase1/chunks/0.1.1.0x061a61d29fa741f5444f53b038d83421b6bf843f-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.1.1.0x823b3d45f7ec13a7de9f05f8174ef87a0ff37081" \
  "gs://plumoceremonyphase1/chunks/0.1.1.0x823b3d45f7ec13a7de9f05f8174ef87a0ff37081-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.1.10.0x02f084313a0a188fcd1e71aad7025eacc64d7388" \
  "gs://plumoceremonyphase1/chunks/0.1.10.0x02f084313a0a188fcd1e71aad7025eacc64d7388-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.1.11.0xf7c4330068e4ebaac4bb11d261b5219074cbdac8" \
  "gs://plumoceremonyphase1/chunks/0.1.11.0xf7c4330068e4ebaac4bb11d261b5219074cbdac8-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.1.12.0x44218a24cfdf0dba912ee5789d8de7a9382739b7" \
  "gs://plumoceremonyphase1/chunks/0.1.12.0x44218a24cfdf0dba912ee5789d8de7a9382739b7-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.1.2.0x44218a24cfdf0dba912ee5789d8de7a9382739b7" \
  "gs://plumoceremonyphase1/chunks/0.1.2.0x44218a24cfdf0dba912ee5789d8de7a9382739b7-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.1.2.0x6bdb9b87845e13dcfe6111d24b7c1c2cc4b7c367" \
  "gs://plumoceremonyphase1/chunks/0.1.2.0x6bdb9b87845e13dcfe6111d24b7c1c2cc4b7c367-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.1.3.0x16a3e97b7b3c04d885b15ff01c346349f936caba" \
  "gs://plumoceremonyphase1/chunks/0.1.3.0x16a3e97b7b3c04d885b15ff01c346349f936caba-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.1.4.0xb1a0d4cc4ee046a016d400f86d3f9036a6f6b192" \
  "gs://plumoceremonyphase1/chunks/0.1.4.0xb1a0d4cc4ee046a016d400f86d3f9036a6f6b192-unsafe" \
  "gs://plumoceremonyphase1/chunks/0.1.5.0x3d776077c2ce34e258ab871a47698ec9a2c81251" \
  "gs://plumoceremonyphase1/chunks/0.1.5.0x3d776077c2ce34e258ab871a47698ec9a2c81251-unsafe" \
  .
```

**Reply by yelhousni** [reply]
For Aleo:
- *Inner* is the Groth16 setup on BLS12-377 (2^19 points),
- *outer* is the Groth16 setup on BW6-761 (2^20 points) and
- *universal* is the Marlin setup on BLS12-377 (2^26 points),

I believe.

**Reply by patrickmao1** [reply]
@yelhousni For Aleo's universal setup, do you happen to know where I can find their encoding spec? I tried to find how I should deserialize the contribution files in their rust code, but it's kinda hard to make out the exact format. I also asked this in their repo but haven't got an answer yet so I decided to try my luck here instead.

**Reply by yelhousni** [reply]
I have the exact same problem :) I want to deserialize the points and convert them to a gnark-compatible format but there is no spec I could find. I am in touch with Alex Pruden (Aleo CEO) and Howard Wu (Aleo CTO) and asked them for help, or to get in touch with their infra team or just to have a spec but I didn’t hear back yet :) It’s crazy that a « publicly verifiable » setup cannot be easily publicly verified. If you manage to make things work let me know, and I will do the same on my end.

**Reply by secure12** [reply]
These repos might be interesting, although I haven't made it work yet:
https://github.com/celo-org/snark-setup-operator
https://github.com/celo-org/snark-setup-coordinator

**Reply by readygo67** [reply]
@patrickmao93  @yelhousni about Aleo SRS file structure, please refer https://github.com/AleoHQ/snarkVM/blob/testnet3/algorithms/src/polycommit/kzg10/data_structures.rs/UniversalParams<E>.
I guess this is what you need, but I'm not 100% sure。

**Reply by secure12** [reply]
@yelhousni Do you have any news on this?
